.github/workflows/main-branch-workflow.yml

```diff
@@ -4,22 +4,31 @@ on:
     push:
       branches:
         - main
+        - update_cicd
 
 jobs:
   publish-to-docker-hub:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
-      - name: Login to Docker Hub
-        uses: docker/login-action@v3
+
+      - name: Set up AWS CLI with credentials for upload to public ECR
```

Comment:
The docker login action also supports AWS ECR logins, see e.g. https://github.com/openbraininstitute/hpc-resource-provisioner/blob/mountpoints/.github/workflows/release.yaml#L40

Reply:
Cool, I like it!